.github/workflows/release.yml

```diff
@@ -5,82 +5,177 @@ permissions:
 
 on:
   workflow_dispatch:
+    inputs:
+      force:
+        required: false
+        description: 'If non-empty, extra force will be used push this release'
+        default: ''
+      commit:
+        required: false 
+        description: 'Fluvio git commit override (latest `master` by default)'
+        default: ''
+env:
+  USE_COMMIT: ${{ github.event.inputs.commit }}
+  FORCE_RELEASE: ${{ github.event.inputs.force }}
 
 jobs:
+  setup_job:
+    name: Setup workflow
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version_step.outputs.VERSION }}
+      target_sha: ${{ steps.version_step.outputs.GIT_SHA }}
+      latest_tag: ${{ steps.docker_step.outputs.LATEST_TAG }}
+      release_tag: ${{ steps.docker_step.outputs.RELEASE_TAG }}
+    steps:
+      - name: Set target sha and Fluvio version
+        id: version_step
+        run: |
+          if [[ -z "${USE_COMMIT}" ]]; then
+            GITHUB_VERSION=$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.sha }}/VERSION)
+            echo "VERSION=${GITHUB_VERSION}" >> $GITHUB_ENV
```

Comment:
I've started to prefer the style `echo "VAR=value" | tee -a $GITHUB_ENV` so that the value being set is printed to stdout and you can verify it in the logs